src/uu/hashsum/src/hashsum.rs

```diff
@@ -135,114 +204,138 @@ fn detect_algo(
             Box::new(Sha3_512::new()) as Box<dyn Digest>,
             512,
         ),
-        "shake128sum" => match matches.get_one::<usize>("bits") {
-            Some(bits) => (
-                "SHAKE128",
-                Box::new(Shake128::new()) as Box<dyn Digest>,
-                *bits,
+        "shake128sum" => create_shake128(matches),
+        "shake256sum" => create_shake256(matches),
+        _ => create_algorithm_from_flags(matches),
+    };
+    (name, alg, output_bits)
+}
+
+/// Handles the "sha3" flag in the given `ArgMatches`.
+///
+/// # Returns
+///
+/// * `Option<(&'static str, Box<dyn Digest>, usize)>` - If the "sha3" flag is set,
+/// returns a tuple with the algorithm name, a boxed digest instance, and the output
+/// size in bits. Otherwise, returns `None`.
+///
+/// # Panics
+///
+/// * If the "bits" argument is not set or has an invalid value (not 224, 256, 384, or 512).
+fn handle_sha3_flag(matches: &ArgMatches) -> Option<(&'static str, Box<dyn Digest>, usize)> {
+    if matches.get_flag("sha3") {
+        match matches.get_one::<usize>("bits") {
+            Some(224) => Some((
+                "SHA3-224",
+                Box::new(Sha3_224::new()) as Box<dyn Digest>,
+                224,
+            )),
+            Some(256) => Some((
+                "SHA3-256",
+                Box::new(Sha3_256::new()) as Box<dyn Digest>,
+                256,
+            )),
+            Some(384) => Some((
+                "SHA3-384",
+                Box::new(Sha3_384::new()) as Box<dyn Digest>,
+                384,
+            )),
+            Some(512) => Some((
+                "SHA3-512",
+                Box::new(Sha3_512::new()) as Box<dyn Digest>,
+                512,
+            )),
+            Some(_) => crash!(
+                1,
+                "Invalid output size for SHA3 (expected 224, 256, 384, or 512)"
             ),
+            None => crash!(1, "--bits required for SHA3"),
+        }
+    } else {
+        None
+    }
+}
+
+/// Creates a hasher instance based on the command-line flags.
+///
+/// # Arguments
+///
+/// * `matches` - A reference to the `ArgMatches` object containing the command-line arguments.
+///
+/// # Returns
+///
+/// Returns a tuple containing the algorithm name, the hasher instance, and the output length in bits.
+///
+/// # Panics
+///
+/// Panics if multiple hash algorithms are specified or if a required flag is missing.
+fn create_algorithm_from_flags(matches: &ArgMatches) -> (&'static str, Box<dyn Digest>, usize) {
```

Comment:
For consistency reasons I would either rename this function to `create_algo_from_flags` or rename the existing function `detect_algo` to `detect_algorithm`